pkg/server/datastore/sqldriver/awsrds/auth_token.go

```diff
@@ -0,0 +1,103 @@
+package awsrds
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/url"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/feature/rds/auth"
+)
+
+type authTokenBuilder interface {
+	buildAuthToken(ctx context.Context, endpoint string, region string, dbUser string, creds aws.CredentialsProvider, optFns ...func(options *auth.BuildAuthTokenOptions)) (string, error)
+}
+
+type tokenGetter interface {
+	getAWSAuthToken(ctx context.Context, params *Config, tokenBuilder authTokenBuilder) (string, error)
+	setNowFunc(nowFunc func() time.Time)
+}
+
+type authToken struct {
+	authToken string
+	expiresAt time.Time
+	nowFunc   func() time.Time
+}
+
+func (a *authToken) getAWSAuthToken(ctx context.Context, config *Config, tokenBuilder authTokenBuilder) (string, error) {
+	if config == nil {
+		return "", errors.New("missing config")
+	}
+
+	if tokenBuilder == nil {
+		return "", errors.New("missing token builder")
+	}
+
+	if !a.isExpired() {
+		return a.authToken, nil
+	}
+
+	awsClientConfig, err := newAWSClientConfig(ctx, config)
+	if err != nil {
+		return "", fmt.Errorf("failed to create AWS Config: %w", err)
+	}
+
+	authenticationToken, err := tokenBuilder.buildAuthToken(ctx, config.Endpoint,
+		config.Region,
+		config.DbUser,
+		awsClientConfig.Credentials)
+	if err != nil {
+		return "", fmt.Errorf("failed to build authentication token: %w", err)
+	}
+
+	values, err := url.ParseQuery(authenticationToken)
+	if err != nil {
+		return "", fmt.Errorf("failed to parse authentication token: %w", err)
+	}
+
+	dateValues := values["X-Amz-Date"]
+	if len(dateValues) != 1 {
+		return "", errors.New("malformed token: could not get X-Amz-Date value")
+	}
+
+	dateTime, err := time.Parse(iso8601BasicFormat, dateValues[0])
+	if err != nil {
+		return "", fmt.Errorf("failed to parse X-Amz-Date date: %w", err)
```

Comment:
`time.Parse` already includes the value as part of the error chain, so if we also include the value in our side, it would look something like this for an "invalid" value, where it would be repeated one more time: `could not get authentication token: failed to parse X-Amz-Date "invalid": parsing time "invalid" as "20060102T150405Z": cannot parse "invalid" as "2006"`